### PR TITLE
Verify the apfs volume doesn't already exist before trying to create it

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

During the plan phase, determine if a Mac APFS volume exists, and error if so with some remedy.

I think @9999years bumped into this in https://github.com/DeterminateSystems/nix-installer/issues/213#issuecomment-1407298341.

##### Checklist

- [x] Formatted with `cargo fmt`
- [x] Built with `nix build`
- [x] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: src/action/darwin/create_apfs_volume.rs [hunk 1]
```diff
@@ -22,6 +22,19 @@ impl CreateApfsVolume {
         name: String,
         case_sensitive: bool,
     ) -> Result<StatefulAction<Self>, ActionError> {
+        let output = execute_command(Command::new("/usr/sbin/diskutil").args(["apfs", "list"]))
```
Comment (9999years) [inline src/action/darwin/create_apfs_volume.rs:25]: This is probably fine, but I'll note that you can run `diskutil apfs list -plist` to get the output as plist XML, and then `/usr/bin/plutil -convert json -o - -` to convert to JSON for easy serde-ability.

Comment (9999years) [reply src/action/darwin/create_apfs_volume.rs:25]: <details>
<summary>
Example output from <code>diskutil apfs list -plist | plutil -convert json -o - -</code>:
</summary>

```json
{
  "Containers": [
    {
      "Fusion": false,
      "PhysicalStores": [
        {
          "DiskUUID": "AFC86EB4-AFBE-4FAE-9210-C3AD26923C8E",
          "DeviceIdentifier": "disk0s1",
          "Size": 524288000
        }
      ],
      "APFSContainerUUID": "66A19B59-AEAC-4EF8-BEAB-00EEA7123921",
      "ContainerReference": "disk1",
      "CapacityCeiling": 524288000,
      "Volumes": [
        {
          "Locked": false,
          "APFSVolumeUUID": "20DCC693-A495-4DE6-B455-462EDA335990",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk1s1",
          "Roles": [
            "Preboot"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "iSCPreboot",
          "Encryption": false,
          "CapacityInUse": 6279168,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "073E5C23-9505-4669-928E-EE00D64B26FB",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk1s2",
          "Roles": [
            "xART"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "xART",
          "Encryption": false,
          "CapacityInUse": 6311936,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "5DDD662F-160F-4ED4-B5E8-B7EBD1645E6F",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk1s3",
          "Roles": [
            "Hardware"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Hardware",
          "Encryption": false,
          "CapacityInUse": 274432,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "9C6D9824-4FBE-44F9-A32E-7B49A3572D49",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk1s4",
          "Roles": [
            "Recovery"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Recovery",
          "Encryption": false,
          "CapacityInUse": 20480,
          "FileVault": false
        }
      ],
      "CapacityFree": 506445824,
      "DesignatedPhysicalStore": "disk0s1"
    },
    {
      "Fusion": false,
      "PhysicalStores": [
        {
          "DiskUUID": "CEB5E52D-E09E-4E1A-AA2B-8E4F2FEB5107",
          "DeviceIdentifier": "disk0s3",
          "Size": 5368664064
        }
      ],
      "APFSContainerUUID": "EDD4837B-9760-4E9C-802D-A565A914B7E0",
      "ContainerReference": "disk2",
      "CapacityCeiling": 5368664064,
      "Volumes": [
        {
          "Locked": false,
          "APFSVolumeUUID": "C5FE138F-1771-4862-B460-CCB7B3038C1E",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk2s1",
          "Roles": [
            "Recovery"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Recovery",
          "Encryption": false,
          "CapacityInUse": 1643745280,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "66BC9C78-7157-48BA-978B-48B72E7FAE22",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk2s2",
          "Roles": [
            "Update"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Update",
          "Encryption": false,
          "CapacityInUse": 176128,
          "FileVault": false
        }
      ],
      "CapacityFree": 3704995840,
      "DesignatedPhysicalStore": "disk0s3"
    },
    {
      "Fusion": false,
      "PhysicalStores": [
        {
          "DiskUUID": "C91C2E32-8A18-42EC-9688-8E07364BDFB8",
          "DeviceIdentifier": "disk0s2",
          "Size": 994662584320
        }
      ],
      "APFSContainerUUID": "D2CA3448-070B-4F22-821A-D3D4805E7745",
      "ContainerReference": "disk3",
      "CapacityCeiling": 994662584320,
      "Volumes": [
        {
          "Locked": false,
          "APFSVolumeUUID": "BCDEEF3E-3A8D-4513-9044-3B49C561A488",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s1",
          "Roles": [
            "System"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Macintosh HD",
          "Encryption": true,
          "CapacityInUse": 8948932608,
          "FileVault": true
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "FB5820C2-EFF7-4065-94B0-4E5271F09E6D",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s2",
          "Roles": [
            "Preboot"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Preboot",
          "Encryption": false,
          "CapacityInUse": 4877594624,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "F073642F-9523-4566-BBD8-C45D4DC0A9F3",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s3",
          "Roles": [
            "Recovery"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Recovery",
          "Encryption": false,
          "CapacityInUse": 785440768,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "3A572134-5E17-4EC3-AEE4-69E2BE9AD9F7",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s4",
          "Roles": [
            "Update"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Update",
          "Encryption": false,
          "CapacityInUse": 13852672,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "73A51853-F5AD-4F08-BD09-33F611EE3FEE",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s5",
          "Roles": [
            "Data"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Data",
          "Encryption": true,
          "CapacityInUse": 212250189824,
          "FileVault": true
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "AD43C339-4438-4FE7-BE34-8BE55CD41C4E",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s6",
          "Roles": [
            "VM"
          ],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "VM",
          "Encryption": false,
          "CapacityInUse": 20480,
          "FileVault": false
        },
        {
          "Locked": false,
          "APFSVolumeUUID": "5D0FC7D6-82F0-4BBA-9FB4-B43AE1CF1332",
          "CapacityQuota": 0,
          "DeviceIdentifier": "disk3s7",
          "Roles": [],
          "CapacityReserve": 0,
          "CryptoMigrationOn": false,
          "Name": "Nix Store",
          "Encryption": true,
          "CapacityInUse": 74899468288,
          "FileVault": true
        }
      ],
      "CapacityFree": 692680396800,
      "DesignatedPhysicalStore": "disk0s2"
    }
  ]
}
```

</details>

Comment (Hoverbear) [reply src/action/darwin/create_apfs_volume.rs:25]: Cool trick! This lets us take plists directly: https://docs.rs/plist/latest/plist/

I can make that change.

File: src/action/darwin/enable_ownership.rs [hunk 2]
```diff
@@ -97,4 +97,7 @@ impl Action for EnableOwnership {
 pub enum EnableOwnershipError {
     #[error("Failed to execute command")]
     Command(#[source] std::io::Error),
+    /// A MacOS (Darwin) plist related error
```
Comment (Hoverbear) [inline src/action/darwin/enable_ownership.rs:100]: Oh I didn't commit an unwrap I spotted and wanted to remove...

**Review by cole-h — APPROVED** [review]
🎉

**Pull request merged**